Ihor/Netflix/src/HomePage/SearchPanel/SearchPanel.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import "./SearchPanel.css";
+import { parse } from "query-string";
+import * as H from "history";
+
+interface ISearchPanelProps {
+  handleSearchChange({
+    search,
+    searchBy,
+  }: {
+    search: string;
+    searchBy: string;
+  }): void;
+  location: H.Location;
+}
+
+interface ISearchPanelState {
+  value: string;
+  searchBy: string;
+}
+
+class SearchPanel extends React.Component<
+  ISearchPanelProps,
+  ISearchPanelState
+> {
+  state: ISearchPanelState = {
+    value: "",
+    searchBy: "title",
+  };
+
+  componentDidMount() {
+    const query = parse(this.props.location.search) as {
+      searchBy: string;
+      search: string;
+    };
+    let { searchBy, search } = query;
+    searchBy ? this.setState({ searchBy: searchBy }) : (searchBy = "title");
+    search ? this.setState({ value: search }) : (search = "");
+  }
+
+  handleSearchParams = (value: string) => {
+    this.setState({ searchBy: value });
+  };
+
+  handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      value: e.target.value,
+    });
+  };
+
+  handleSubmit = () => {
+    this.setState({ value: "" });
+    this.props.handleSearchChange({
+      search: this.state.value,
+      searchBy: this.state.searchBy,
+    });
+  };
+
+  keyPressOn = (e: React.KeyboardEvent) => {
+    if (e.key === "Enter") {
+      this.handleSubmit();
+    }
+    return this.handleChangeInput, this.handleSearchParams;
```

Comment:
why ? what do you expect of it? have you ever seen several functions in return statement ??